hikari/utilities/iterators.py

```diff
@@ -497,19 +516,32 @@ async def last(self) -> ValueT:
 
         Returns
         -------
-        T
+        ValueT
             The last result.
 
+        !!! note
+            This method will deplete the iterator if run.
+
         Raises
         ------
         builtins.LookupError
             If no result exists.
         """
         try:
-            return (await self)[0]
+            return (await self.reversed())[0]
         except IndexError:
             raise LookupError("No elements were found") from None
 
+    async def reversed(self) -> typing.Sequence[ValueT]:
```

Comment:
Can we return a LazyIterator instead of a sequence here.

Reason I ask is that we can then chain stuff onto this.

Reply:
Well it won't be lazy anymore but sure

Reply:
It can be, if the implementation overrides this.

For example a paginated endpoint could just theoretically change the call it was making if the info were available.

Reply:
also if those implementations overrode it they'd have to track the point where it'd stop otherwise they'd be changing the behaviour from the reminder of the iterator's values to the iterator's values but reversed

Reply:
the point is that the implementation detail remains consistent and doesn't impose a full implementation earlier than needed.

This way you can get the last 5 items using the same syntax: `async for item in it.reversed().limit(5):` thus remaining consistent.

If they want to just operate on a reversed list directly they should just run `list(reversed(await it))` to get that eagerly.

If you change this to a synchronous endpoint `def reverse()` that returns a lazy iterator, they can still await it as the same syntax to get the internal result, or they can continue to operate on it elsewhere.

Especially when I implement stuff like flatMap, this will be much more useful for streaming manipulations.